src/dmd/expressionsem.d

```diff
@@ -6626,6 +6626,14 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
 
             exp.type = exp.type.addMod(t1.mod);
 
+            // https://issues.dlang.org/show_bug.cgi?id=23109
+            // Run semantic on the DotVarExp type
+            if (auto handle = exp.type.isClassHandle())
+            {
+                if (handle.semanticRun < PASS.semanticdone && !handle.isBaseInfoComplete())
+                    handle.dsymbolSemantic(null);
+            }
+
```

Comment:
I take it the problem is https://github.com/dlang/dmd/blob/8cea13cd302bba4a7fd1dd25c81e19ad8682daaa/src/dmd/expressionsem.d#L1076 in `getRightThis()`? - Wouldn't it be safer to move it there? Or are you concerned about performance implications?

Reply:
No, the immediate context where `isBaseOf` is called from is `TypeClass::deduceWild`, and the DotVarExp is constructed from `TypeStruct::dotExp(ident="tupleof")`.

In the fix-ups I've done so far, after changing `isBaseOf` to pure, I've just put it in the semantic routines after the point where type is no longer accessed.  Don't really want to call dsymbolSemantic too early, else problems may ensue.